app/src/main/java/com/cmput301f20t21/bookfriends/ui/login/CreateAccountActivity.java

```diff
@@ -77,58 +81,104 @@ public void afterTextChanged(Editable emailString) {
     }
 
     /**
-     * check the input field for emptiness, adding error message if field is empty
-     * @param layout the input field layout that the user entered
-     * @return true if the field is empty, false if field is not empty
+     * back to the login in page when clicking cancel button
+     * @param view
      */
-    private boolean checkEmpty(TextInputLayout layout) {
-        EditText inputField = layout.getEditText();
-        String userInput  = inputField.getText().toString();
-        String inputHint = inputField.getHint().toString();
-        boolean isFieldEmpty = userInput.isEmpty();
-
-        if (isFieldEmpty) {
-            layout.setError( inputHint + getString(R.string.empty_error));
-        }
-        return isFieldEmpty;
-    }
-
     public void onCancelClicked(View view) {
         finish();
     }
 
+    /**
+     * create the account if there's no input errors, or display errors
+     * @param view
+     */
     public void onCreateClicked(View view){
-        String confirmPassword = confirmField.getText().toString();
-        String password = passwordField.getText().toString();
-        if(!confirmPassword.equals(password)){
+        boolean isEmailValid = model.isEmailValid(emailLayout);
+        boolean correctPasswordLength = model.correctPasswordLength(passwordLayout);
+        boolean isUsernameValid = model.isUsernameValid(usernameLayout);
+        boolean usernameNoSpace = model.noSpace(usernameLayout);
+        boolean passwordNoSpace = model.noSpace(passwordLayout);
+        boolean usernameNotEmpty = model.notEmpty(usernameLayout);
+        boolean confirmNotEmpty = model.notEmpty(confirmLayout);
+        boolean passwordNotEmpty = model.notEmpty(passwordLayout);
+        boolean emailNotEmpty = model.notEmpty(emailLayout);
+        boolean isPasswordMatch = model.isPasswordMatch( passwordLayout, confirmLayout);
```

Comment:
I think we don't have to declare all these variable since model functions will be descriptive enough.
We can just do: `if (model.isPasswordMatch( passwordLayout, confirmLayout)) {...}
Same for all above.